modules/benchmark/R/create.BRR.R

```diff
@@ -9,48 +9,49 @@
-  cnd1 <- workflow$hostname == fqdn()
-  cnd2 <- workflow$hostname == "test-pecan.bu.edu" & fqdn() == "pecan2.bu.edu"
-  cnd3 <- workflow$hostname == "pecan2.bu.edu" & fqdn() == "test-pecan.bu.edu"
-
-  if (cnd1 | cnd2 | cnd3) {
-    # If the ensemble run was done on localhost, turn into a BRR
+  cnd1 <- ens_wf$hostname == fqdn() 
+  cnd2 <- ens_wf$hostname == 'test-pecan.bu.edu' & fqdn() == 'pecan2.bu.edu'
+  cnd3 <- ens_wf$hostname == 'pecan2.bu.edu' & fqdn() == 'test-pecan.bu.edu'
+  
+  
+  if(cnd1|cnd2|cnd3){  # If the ensemble run was done on localhost, turn into a BRR
     
-    BRR <- db.query(paste0("INSERT INTO reference_runs (model_id, settings, user_id, created_at, updated_at) VALUES(", 
-      workflow$model_id, ", '", workflow$folder, "' , ", user_id, ", NOW() , NOW()) RETURNING *;"), 
-      con)
+    settingsXML <- file.path(ens_wf$folder,"pecan.CHECKED.xml")
     
-    bm.ensemble <- db.query(paste0("INSERT INTO benchmarks_ensembles (reference_run_id, ensemble_id, model_id, user_id, created_at, updated_at, citation_id) VALUES(", 
-      BRR$id, ",", ensemble.id, ",", BRR$model_id, ", ", user_id, ", NOW() , NOW(), 1000000001 ) RETURNING *;"), 
-      con)
+    # Automatically creates a new pecan.xml I think. Need to fix this. 
+    clean <- clean.settings(settingsXML, "findme.xml")
+    # Remove database & host information
+    clean$database <- NULL 
+    clean$host <- NULL
+    clean$info <- NULL
+    clean$outdir <- NULL
+    str(clean)
     
-    return(BRR)
-  } else {
-    logger.error(sprintf("Cannot create a benchmark reference run for a run on hostname: %s", 
-      workflow$hostname))
-  }
-} # create.BRR
-
-
-# Case in which we would need a remote connection to get the pecan.xml file - not functional
+    settings_xml <- toString(listToXml(clean, "pecan"))
+    
+    ref_run <- db.query(paste0(" SELECT * from reference_runs where settings = '", settings_xml,"'"),con)
+    
+    if(length(ref_run) == 0){ # Make new reference run entry
+      ref_run <- db.query(paste0("INSERT INTO reference_runs",
+                                 "(model_id, settings, user_id, created_at, updated_at)",
+                                 "VALUES(",ens_wf$model_id,", '",settings_xml,"' , ",user_id,
+                                 ", NOW() , NOW()) RETURNING *;"),con)
+    }else if(dim(ref_run)[1] > 1){# There shouldn't be more than one reference run with the same settings
+      logger.error("There is more than one reference run in the database with these settings. Review for duplicates. ")
+    }
+    
+    bm.ensemble <- db.query(paste0("INSERT INTO benchmarks_ensembles",
+                                   "(reference_run_id, ensemble_id, model_id, ",
+                                   "user_id, created_at, updated_at, citation_id)",
+                                   "VALUES(",ref_run$id,",",ens_wf$ensemble_id,",", ref_run$model_id,", ",user_id,
+                                   ", NOW() , NOW(), 1000000001 ) RETURNING *;"),con)
```

Comment:
1) would be good to also check for duplicates here
2) as above, drop NOW() 
3) why are you inserting citation_id 1000000001 as a hard-coded value rather than leaving this blank?  Citation_id is an optional value, and I suspect that it will often be added when the results are published, not when they're created. Maybe make it an optional argument to the function, and one that defaults to blank or NULL?